### ISSUE TITLE
IPv6 and dual stack Azure tests failing to pull image with k8s HEAD

### ISSUE THREAD

**Issue by CecileRobertMichon** [kubernetes-sigs/cluster-api-provider-azure#4180]
**Which jobs are failing:**
https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-pr-conformance-dual-stack-k8s-ci-main
https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-pr-conformance-ipv6-k8s-ci-main
https://testgrid.k8s.io/provider-azure-master-signal#capz-conformance-ipv6
https://testgrid.k8s.io/provider-azure-master-signal#capz-conformance-dual-stack

**Which tests are failing:**

Calico tigera install pod is failing to start with

```
  Warning  Failed     16m (x4 over 18m)    kubelet            Error: ErrImagePull
  Warning  Failed     16m                  kubelet            Failed to pull image "mcr.microsoft.com/oss/tigera/operator:v1.30.4": failed to pull and unpack image "mcr.microsoft.com/oss/tigera/operator:v1.30.4": failed to resolve reference "mcr.microsoft.com/oss/tigera/operator:v1.30.4": failed to do request: Head "https://mcr.microsoft.com/v2/oss/tigera/operator/manifests/v1.30.4": dial tcp: lookup mcr.microsoft.com on 127.0.0.53:53: read udp 127.0.0.1:49168->127.0.0.53:53: read: connection refused``
```

**Since when has it been failing:** 10/24

**Testgrid link:**

**Reason for failure (if possible):**

**Anything else we need to know:**

/kind failing-test

[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by CecileRobertMichon** [comment]
the release-1.28 branch jobs are passing so this is/was only an issue with k8s main branch
[provider-azure-1.28-signal Test Grid (k8s.io)](https://testgrid.k8s.io/provider-azure-1.28-signal#capz-conformance-ipv6)

**Comment by CecileRobertMichon** [comment]
/close

this was fixed in #4209
